includes/Product/Product.php

```diff
@@ -0,0 +1,189 @@
+<?php
+/**
+ * Class Product
+ *
+ * @link      https://github.com/googleforcreators/web-stories-wp
+ *
+ * @copyright 2022 Google LLC
+ * @license   https://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ */
+
+/**
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Web_Stories\Product;
+
+/**
+ * Class Product
+ */
+class Product {
+	/**
+	 * Product id.
+	 *
+	 * @var string
+	 */
+	protected $id = '';
+	/**
+	 * Product title.
+	 *
+	 * @var string
+	 */
+	protected $title = '';
+	/**
+	 * Product brand.
+	 *
+	 * @var string
+	 */
+	protected $brand = '';
+	/**
+	 * Product Price.
+	 *
+	 * @var float
+	 */
+	protected $price = 0.0;
+	/**
+	 * Product's price currency.
+	 *
+	 * @var string
+	 */
+	protected $price_currency = '';
+	/**
+	 * Product images as an array.
+	 *
+	 * @var array
+	 */
+	protected $images = [];
+	/**
+	 * Product Details.
+	 *
+	 * @var string
+	 */
+	protected $details = '';
+	/**
+	 * Product url.
+	 *
+	 * @var string
+	 */
+	protected $url = '';
+	/**
+	 * Product rating.
+	 *
+	 * @var array
+	 */
+	protected $aggregate_rating = [
+		'rating_value' => 0.0,
+		'review_count' => 0,
+		'review_url'   => '',
+	];
+
+	/**
+	 * Product constructor.
+	 *
+	 * @since 1.21.0
+	 *
+	 * @param array $product Array of attributes.
+	 */
+	public function __construct( array $product = [] ) {
+		foreach ( $product as $key => $value ) {
+			if ( property_exists( $this, $key ) ) {
+				$this->$key = $value;
+			}
+		}
+	}
```

Comment:
We can add validation of properties later to ensure that typing and values are correct. But in this model is where I would expect that to happen.